### ISSUE TITLE
Possible to have one api file for the whole api?

### ISSUE THREAD

**Issue by Dashue** [hoangvvo/next-connect#53]
I was looking into this as a means of cutting down on the lambdas due to vercels new pricing structure.

Is it possible to have a single api file "api/index.ts" that is used to map all the routes?
Or is it restricted to one api per resource user/index.ts, product/index.ts etc?

Thanks

**Comment by Dashue** [comment]
Oh, think I've found a solution using catch all api approach.

https://nextjs.org/docs/api-routes/dynamic-api-routes#catch-all-api-routes

**Comment by Dashue** [comment]
@hoangvvo I was able to make this work except catching query parameters.

I have a single api endpoint
pages/api/[...param].ts

Handler setup:
.use('/api/sitemap', SiteMapApi)

Logging of handler shows path
 {
      keys: [],
      pattern: /^\/api\/sitemap(?=$|\/)/i,
      method: '',
      handlers: [Array]
    },

Request
http://localhost:3000/api/sitemap?type=studios

Output
Error: Route didn't match {"type":"studios","param":["sitemap"]}

const onNoMatch = (req: NextApiRequest, res: NextApiResponse) => {
    LogError(new Error(`Route didn't match ${JSON.stringify(req.query)}`))
    res.status(404).end();
}

I found that if I add a / before the type query parameter it works
http://localhost:3000/api/sitemap/?type=studios

**Comment by hoangvvo** [comment]
This is odd. I will take a look at it.

**Comment by hoangvvo** [comment]
Any chance you can create a minimal reproduction for me?

**Comment by MatthieuCoelho** [comment]
Same error

To reproduce, create a file:
[...params].ts in /api

```
export default nextConnect()
      .use("/api/locations",(req, res, next) => {
          const {query: {search}, method} = req;
          console.log(search);  
          next();
    });
```
http://localhost:3000/api/locations => undefined
http://localhost:3000/api/locations/?search=toto => toto
http://localhost:3000/api/locations?search=toto => Route does not match.

**Comment by hoangvvo** [comment]
@Dashue @MatthieuCoelho Sorry for delay everyone! It is resolved in `0.7.1`.

Thanks @Dashue for suggesting this use case. This really helps teams migrating from `express` without rewriting all the .js/.ts in `api`.
